sim/src/lib.rs

```diff
@@ -362,29 +362,30 @@ impl SidewalkSpot {
             });
         }
 
-        let lanes = map
+        map
             .get_i(i)
-            .get_incoming_lanes(map, PathConstraints::Pedestrian);
-        if lanes.is_empty() {
-            return None;
-        }
-        Some(SidewalkSpot {
-            sidewalk_pos: Position::new(lanes[0], map.get_l(lanes[0]).length()),
-            connection: SidewalkPOI::Border(i, origin),
-        })
+            .get_incoming_lanes(map, PathConstraints::Pedestrian)
+            .next()
+            .map(|lane0| {
```

Comment:
nit: for consistency with the rest of the code, I'd prefer just `l` for the var name here. I think from the `.next()` call it's clear enough that this is the first result